extensions/panache/hibernate-orm-panache/runtime/src/main/java/io/quarkus/hibernate/orm/panache/runtime/JpaOperations.java

```diff
@@ -249,7 +260,12 @@ public static Object findById(Class<?> entityClass, Object id, LockModeType lock
         String findQuery = createFindQuery(entityClass, query, paramCount(params));
         EntityManager em = getEntityManager();
         // FIXME: check for duplicate ORDER BY clause?
-        Query jpaQuery = em.createQuery(sort != null ? findQuery + toOrderBy(sort) : findQuery);
+        Query jpaQuery;
+        if (isNamedQuery(query)) {
```

Comment:
Well, it can also happen here. You could have fixed `isNamedQuery` instead ;)

Reply:
No, here this is the query computed by `createFindQuery` and it's never null (if null it will be `from EntityClass`.

Reply:
it looks like the query created by `createFindQuery` is stored in `findQuery` and not in the `query` used, which is a parameter.

Reply:
Nope, this comes straight from the user.

Reply:
Damned it! I need to clean my glasses ;)
I will add a test with null and emtpy query so we didn't risk such bugs later

Reply:
@FroMage fixed.
I also add tests for empty and null query.